### PR TITLE
fix: Karpenter eventbridge naming

### LINKED ISSUE #1482: [Karpenter] Rule with the name already exists on this event bus.
## Description

We cannot create Event bridge rules when we deploy more than one cluster in the same region with this option. `karpenter_enable_spot_termination_handling = true`

## Versions

- Module version [Required]:
v4.25.0
- Terraform version:
v1.3.9

- Provider version(s):
```
+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/hashicorp/aws v4.53.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/helm v2.8.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.17.0
+ provider registry.terraform.io/hashicorp/local v2.3.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/tls v4.0.4
+ provider registry.terraform.io/terraform-aws-modules/http v2.4.1
```

## Reproduction Code [Required]

https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/e0e9aeb7ea785613cf794d524f79bcf5929bb6b8/modules/kubernetes-addons/karpenter/main.tf#L40

## Expected behavior

should be able to create more than one cluster in the same region without conflict with the resources naming

## Actual behavior
`Rule with the name already exists on this event bus.`

### PR DESCRIPTION
**Pull request by dtherhtun**
## :bangbang: PLEASE READ THIS FIRST :bangbang:

The direction for EKS Blueprints will soon shift from providing an all-encompassing, monolithic "framework" and instead focus more on how users can organize a set of modular components to create the desired solution on Amazon EKS. We have updated the [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to show how we use the https://github.com/terraform-aws-modules/terraform-aws-eks for EKS cluster and node group creation. We will not be accepting any PRs that apply to EKS cluster or node group creation process. Any such PR may be closed by the maintainers.

We are hitting also the pause button on new add-on creations at this time until a future roadmap for add-ons is finalized. Please do not submit new add-on PRs. Any such PR may be closed by the maintainers.

Please track progress, learn what's new and how the migration path would look like to upgrade your current Terraform deployments. We welcome the EKS Blueprints community to continue the discussion in issue https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1421

### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

- Resolves #1482 #1471

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by dtherhtun** [comment]
Screenshot 2023-03-07 at 18 23 42
- Tested

**Review by vara-bonthu — APPROVED** [review]
LGTM 👍🏼

File: modules/kubernetes-addons/karpenter/main.tf [hunk 1]
```diff
@@ -37,7 +37,7 @@ resource "aws_sqs_queue_policy" "this" {
 resource "aws_cloudwatch_event_rule" "this" {
   for_each = { for k, v in local.event_rules : k => v if var.enable_spot_termination }
 
-  name          = each.value.name
+  name          = join("-", [each.value.name, var.addon_context.eks_cluster_id])
```
Comment (askulkarni2) [inline modules/kubernetes-addons/karpenter/main.tf:40]: I would prefer using the `name_prefix` over a fixed `name` here. Similar to https://github.com/terraform-aws-modules/terraform-aws-eks/blob/c7565e265e23cbc622041fc153193f490cfe948f/modules/karpenter/main.tf#L270.

```suggestion
  name_prefix   =  each.value.name
```

**Comment by askulkarni2** [comment]
@dtherhtun can you please re-open this PR against https://github.com/aws-ia/terraform-aws-eks-blueprints-addons repo? This is going to be the new home for blueprints addons repo going forward as a part of changes described in #1421.

**Pull request closed without merging**